Features #161089368, #162396232 – Homepage redesign

alfonsomunozpomer opened the pull request:
Quite a few changes, so I’ll try to summarise the main points here:
• Rename `PopularSpecies`-whatever to `SpeciesSummary`-whatever. I’ve never been too happy with that name and it serves us to, maybe in the future, get information about specific species.
• Lift the renamed `SpeciesSummary` controller logic to `atlas-core` so it can be reused by bulk and SC, changing the necessary bits by injecting a different DAO. The response is now different and the results are always grouped by kingdom. In this way the React component that displays the species is just presentational and doesn’t need to know anything about kingdom values.
• Remove the popular species fields from the model in the homepage controller.
• Add a new repo, [`species-summary-panel`](https://github.com/ebi-gene-expression-group/species-summary-panel), as a git submodule in the JS modules directory, which is bundled with [`atlas-react-fetch-loader`](https://github.com/ebi-gene-expression-group/atlas-react-fetch-loader). The bundle makes a request to the species-summary endpoint and displays the species in tabs. It also relies on the [latest version of `atlas-homepage-cards`](https://github.com/ebi-gene-expression-group/atlas-homepage-cards/pull/11).
• Follow a similar approach for latest and featured experiments, so that they are served from a single endpoint. The shame here is that, unlike species, the field `latestExperiments` can’t be serialised as cards. Maybe something we can think a bit better and imporve: if we do, we can refactor [`species-summary-panel`](https://github.com/ebi-gene-expression-group/species-summary-panel) and [`experiments-summary-panel`](https://github.com/ebi-gene-expression-group/experiments-summary-panel) to a single component.

Worthy of note is that I made the `description.text` field in the card model optional, since the featured experiments don’t have it. @monicajianu could you please change the JSON schema to reflect that?

Other improvements:
• Remove the hero image+text in SCEA.
• Use Foundation classes to lay out the release stats at the top. Remember that we shouldn’t use flex or grid and rely on the EBI VF if it’s in the environment (widgets embeddable in third parties are the exception but we have one or two of those).

alfonsomunozpomer committed:
Remove popular species from the home controller
We’re no longer adding them to the model but reading them from a dedicated JSON controller that the React component requests

alfonsomunozpomer committed:
Rename PopularSpeciesInfo to SpeciesSummary and add a common controller implementation

alfonsomunozpomer committed:
Make controllers use the implementation from atlas-core and test new format of JSON response

alfonsomunozpomer committed:
Make code more functional

alfonsomunozpomer committed:
Remove hero from home view

alfonsomunozpomer committed:
Add Atlas Homepage Species Summary Panel component as a module to both EA and SCEA

alfonsomunozpomer committed:
Move atlas-homepage-species-summary-panel to modules

alfonsomunozpomer committed:
Rename and update species-summary-panel

alfonsomunozpomer committed:
Update submodule species-summary-panel

alfonsomunozpomer committed:
Rename submodule entries

alfonsomunozpomer committed:
Make both fields in description non-required to adjust the card model to featured experiments

alfonsomunozpomer committed:
Add new JSON endpoint experiments-summary that provides both featured and latest experiments to the new experiments-summary-panel component

alfonsomunozpomer committed:
Remove unnecessary bundles

alfonsomunozpomer committed:
Upate modules to latest revisions

alfonsomunozpomer committed:
Add new bundles that wrap atlas-react-fetch-loader around species-summary-panel and experiments-summary-panel

alfonsomunozpomer committed:
Delete unnecessary bundles

alfonsomunozpomer committed:
Add new bundles that wrap atlas-react-fetch-loader around species-summary-panel, experiments-summary-panel and atlas-homepage-cards

alfonsomunozpomer committed:
Fix controllers’s JSON response to conform to new panel components

alfonsomunozpomer committed:
Bump versions

alfonsomunozpomer committed:
Update dependencies

alfonsomunozpomer committed:
Dummy test to at least ensure that the branch compiles and passes in Bamboo :/

alfonsomunozpomer committed:
Fix test annotations to use JUnit 5 properly

alfonsomunozpomer committed:
Don’t break lines between an item and its count/version

alfonsomunozpomer committed:
Update Download page

alfonsomunozpomer committed:
Rename temp variable to something a bit more meaningful

alfonsomunozpomer committed:
Add proper tests for SpeciesSummaryService

alfonsomunozpomer committed:
Use all available space for the text under the title

alfonsomunozpomer committed:
Update submodules

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/home/SpeciesSummaryService.java:
```diff
@@ -0,0 +1,60 @@
+                        .collect(groupingBy(SpeciesSummary::getKingdom))
+                        .entrySet().stream()
+                        .collect(toImmutableSortedMap(
+                                new ArbitraryStringComparator("animals", "plants", "fungi", "protists"),
```
I don't think sorting of this kind is the responsiblity of the service. I think it's either the UIs responsiblity, or at the very least, the controller.

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/model/card/CardModelAdapter.java:
```diff
@@ -53,4 +51,12 @@ private static JsonObject getTextAndUrl(String text, Optional<String> url) {
         return textAndUrl;
     }
+
+    private static JsonObject getTextAndUrl(Pair<Optional<String>, Optional<String>> description) {
```
Hmm there's a bit of repeated logic here when with `getLeft` and `getRight`. Could you not pass two `Optionals` to this method, the text and the url, rather than the `Pair`? That was it will also be consistent with the other implementation of the `getTextAndUrl` method.

monicajianu commented on gxa/src/main/java/uk/ac/ebi/atlas/home/JsonExperimentsSummaryController.java:
```diff
@@ -0,0 +1,177 @@
+                                .collect(toImmutableList())));
+    }
+
+    private static String getImageUrl(String imageFileName) {
```
These URL methods could be moved to the `UrlHelpers` class.

monicajianu commented on gxa/src/main/webapp/WEB-INF/jsp/tiles/views/home/experiments-summary-panel.jsp:
```diff
@@ -0,0 +1,25 @@
+          imageIconHeight: '3rem'
+        },
+        onComponentDidMount: function() {
+          $('#experiments-summary-panel').foundation();
```
👍

monicajianu commented on gxa/src/main/webapp/WEB-INF/jsp/tiles/views/home/release-stats.jsp:
```diff
@@ -0,0 +1,26 @@
+<%@ page contentType="text/html;charset=UTF-8" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+
+<div class= "small-12 medium-6 columns text-left">
```
I think we could get away with making this text a little smaller. If you view the page on a normal-sized screen (laptop), the text takes up a second line.

<img width="1677" alt="Screenshot 2019-03-22 at 14 32 57" src="https://user-images.githubusercontent.com/5923538/54829936-6a6c9700-4caf-11e9-8707-c859675844bc.png">

alfonsomunozpomer replied:
Yes, I noticed it too when I saw it on my laptop. I think `<h4><small>` will work. Thanks for reminding me.

monicajianu reviewed (changes requested):
A few minor things. I'll also change the JSON schema.

alfonsomunozpomer committed:
Fix layout of release stats: remove column in enclosing div to remove margins, make font small and hide ontology versions in small screens

alfonsomunozpomer committed:
Delay kingdom ordering of species summary cards to the controller

alfonsomunozpomer committed:
Move UrlHelpers to package utils as it’s going to be used by more than one class

alfonsomunozpomer committed:
Make UrlHelpers a class with static methods instead of an interface

alfonsomunozpomer committed:
Refactor logic in UrlHelpers and its clients

alfonsomunozpomer committed:
Add tests to UrlHelpers and change location of featured experiments logos to a more aptly named directory

alfonsomunozpomer committed:
Remove references to previous incarnation of featured experiments panel

alfonsomunozpomer committed:
Expand global masthead with an unsightly hack and align header button to the right

alfonsomunozpomer committed:
Add release stats to SCEA

alfonsomunozpomer committed:
Add button to see all experiments “for all species” below species summary panel

alfonsomunozpomer commented:
@monicajianu I addressed all the issues you mentioned. I especially like how I ended up refactoring `UrlHelpers`. It forced me to rethink things a bit but the result is much better, IMO.

alfonsomunozpomer committed:
Generate production bundles

alfonsomunozpomer committed:
Generate production bundles

monicajianu commented on sc/src/main/webapp/WEB-INF/tiles/views.xml:
```diff
@@ -8,6 +8,7 @@
         <put-attribute name="title" value="Home &lt; "/>
         <put-attribute name="content">
             <definition template="/WEB-INF/jsp/tiles/views/home.jsp">
+                <put-attribute name="rel" value="/WEB-INF/jsp/tiles/views/home/release-stats.jsp"/>
```
Can we give this a nicer, more explicit name, rather than "rel"?

alfonsomunozpomer replied:
Oh, right! Thanks for spotting it.

monicajianu commented on gxa/src/main/webapp/WEB-INF/jsp/tiles/global-masthead.jsp:
```diff
@@ -2,3 +2,10 @@
+
+<script>
+  // Hack that solves https://github.com/ebiwd/EBI-Framework/pull/139 (I got tired of waiting)
+  window.addEventListener('load', function(event) {
```
Hmm the side-effect of this is that there is a delay, so it starts off unexpanded and then you can see it move (or maybe my browser is particularly sluggish today). But I can't think of a better way of doing it...

alfonsomunozpomer replied:
I know, I’m aware of it. Once the assets are cached it’s very fast, although noticeable. It’s the kind of thing that people who are unaware of it will hardly ever notice.

Another option is to fork the framework but then we can’t use the CDN. Let’s hope the PR gets trhough, although it seemed it won’t because it falls outside their core design principles. I still think that if `row expanded` is a supported option, it should also be available on the global masthead. Otherwise pages look kind of unbalanced.

Anyway...

monicajianu reviewed (approved):
One tiny variable name change and then it's good to go. I like the static `UrlHelpers`, I think they used to be like that when I first wrote the class 🙂

alfonsomunozpomer committed:
Rename view attribute

alfonsomunozpomer committed:
Update JS submodules

alfonsomunozpomer committed:
Merge remote-tracking branch 'origin/dev' into feature/161089368-162396232-redesign-home-page

# Conflicts:
#	sc/src/main/javascript/bundles/gene-search/package-lock.json
#	sc/src/main/javascript/bundles/gene-search/package.json
#	sc/src/main/webapp/resources/js-bundles/experimentPage.bundle.js
#	sc/src/main/webapp/resources/js-bundles/feedbackForm.bundle.js
#	sc/src/main/webapp/resources/js-bundles/geneSearch.bundle.js
#	sc/src/main/webapp/resources/js-bundles/homePagePanel.bundle.js
#	sc/src/main/webapp/resources/js-bundles/vendorCommons.bundle.js
